Gems/ROS2/Code/Tests/SdfParserTest.cpp

```diff
@@ -0,0 +1,241 @@
+            const bool success = sdf::readString(xmlString, sdfElement, readErrors);
+            if (!success)
+            {
+                return nullptr;
+            }
+
+            auto sdfDom = AZStd::make_shared<sdf::Root>();
+            sdf::Errors parseErrors = sdfDom->Load(sdfElement);
+            return sdfDom;
+        }
+
+        std::string GetSdfWithTwoSensors()
+        {
+            return "<?xml version=\"1.0\" ?>\n"
+                   "<sdf version=\"1.6\">\n"
+                   "  <model name=\"test_two_sensors\">\n"
+                   "    <link name=\"link1\">\n"
+                   "      <sensor name=\"camera\" type=\"camera\">\n"
+                   "        <pose>0 0 0 0 0 0</pose>\n"
+                   "        <camera>\n"
+                   "          <horizontal_fov>2.0</horizontal_fov>\n"
+                   "          <image>\n"
+                   "            <width>640</width>\n"
+                   "            <height>480</height>\n"
+                   "          </image>\n"
+                   "          <clip>\n"
+                   "            <near>0.01</near>\n"
+                   "            <far>1000</far>\n"
+                   "          </clip>\n"
+                   "        </camera>\n"
+                   "        <update_rate>10</update_rate>\n"
+                   "        <plugin name=\"camera_plug\" filename=\"libgazebo_ros_camera.so\">\n"
+                   "          <camera_name>custom_camera</camera_name>\n"
+                   "        </plugin>\n"
+                   "      </sensor>\n"
+                   "    </link>\n"
+                   "    <link name=\"link2\">\n"
+                   "      <sensor name=\"laser\" type=\"ray\">\n"
+                   "        <always_on>1</always_on>\n"
+                   "        <visualize>1</visualize>\n"
+                   "        <update_rate>20.0</update_rate>\n"
+                   "        <pose>0 0 0 0 0 0</pose>\n"
+                   "        <ray>\n"
+                   "          <scan>\n"
+                   "            <horizontal>\n"
+                   "              <samples>640</samples>\n"
+                   "              <resolution>1.0</resolution>\n"
+                   "              <min_angle>-2.0</min_angle>\n"
+                   "              <max_angle>2.5</max_angle>\n"
+                   "            </horizontal>\n"
+                   "          </scan>\n"
+                   "          <range>\n"
+                   "            <min>0.02</min>\n"
+                   "            <max>10</max>\n"
+                   "            <resolution>0.01</resolution>\n"
+                   "          </range>\n"
+                   "        </ray>\n"
+                   "        <plugin name=\"laser_plug\" filename=\"librayplugin.so\"/>\n"
+                   "      </sensor>\n"
+                   "    </link>\n"
+                   "  </model>\n"
+                   "</sdf>\n";
+        }
```

Comment:
FYI: For future references, if you need support for a multiline string, you can also use the Raw string literal `R"()"` support: https://en.cppreference.com/w/cpp/language/string_literal.

In that case the `'\n'` wouldn't need to be explicitly specified and the double quotes wouldn't need to be escape..
Since XML is being parsed the leading whitespace before tags isn't relevant to the parsing aspect".

For example
```
constexpr std::string_view foo = R"(<ObjectStream version="3">
                    <Class name="DataPatch" type="{3A8D5EC9-D70E-41CB-879C-DEF6A6D6ED03}">
                            <Class name="AZ::Uuid" field="m_targetClassId" value="{47E5CF10-3FA1-4064-BE7A-70E3143B4025}" type="{E152C105-A133-4D03-BBF8-3D4B2FBA3E2A}"/>
                            <Class name="AZStd::unordered_map" field="m_patch" type="{CF3B3C65-49C0-5199-B671-E75347EB25C2}">
                                    <Class name="AZStd::pair" field="element" type="{07DEDB71-0585-5BE6-83FF-1C9029B9E5DB}">
                                            <Class name="AddressType" field="value1" value="C705B33500000000" type="{90752F2D-CBD3-4EE9-9CDD-447E797C8408}"/>
                                            <Class name="ByteStream" field="value2" value="00000000031C72039442EB384D42A1ADCB68F7E0EEF6000000960000" type="{ADFD596B-7177-5519-9752-BC418FE42963}"/>
                                    </Class>
                            </Class>
                    </Class>
            </ObjectStream>
            )";
```